### PR TITLE
Update status page for more complex configs.

### PR DESCRIPTION
**Pull request by plural**
The google playlist has some modules that aren't in a playlist
and can't be played temporarily.  They can only be played indefinitely
since that makes a fake playlist and forces a server reload.

### CODE REVIEW

File: client/status/status.js [hunk 1]
```diff
@@ -20,11 +20,24 @@ var bgColors = [
...
 
+// Returns a map of module name to number.
+// 1 means the module exists in the config.
+// > 1 means the module exists and is in a loaded playlist.
+// Modules that are in the modules list, but not in a playlist aren't eligible
+// for immediate playback, only for indefinite takeover.
 function buildPlayableModuleMap(config) {
-  // which collections are in the playlist?
+  let playableModules = {};
+  config.current.modules.forEach(function(module) {
```
Comment (applmak) [inline client/status/status.js]: module => playableModules[module.name] = 1

File: client/status/status.js [hunk 2]
```diff
@@ -20,11 +20,24 @@ var bgColors = [
...
+    playableModules[module.name] = 1;
+  });
+
   let playableCollections = {};
   config.current.playlist.forEach(function(playlist) {
-    if (playlist.collection == '__ALL__') {
+    if (playlist.collection === undefined) {
+      playlist.modules.forEach(function(module) {
```
Comment (applmak) [inline client/status/status.js]: ditto

File: client/status/status.js [hunk 3]
```diff
@@ -207,15 +219,19 @@ fetchJson('config').then(config => {
   let module_list = document.getElementById('module_list');
   config.current.modules.forEach(function(module) {
     let li = document.createElement("li");
-    let a = document.createElement("a");
-    a.id = 'module_' + module.name;
-    if (module.name in modulesInPlaylist) {
+    if (modulesInPlaylist[module.name] > 1) {
+      let a = document.createElement("a");
```
Comment (applmak) [inline client/status/status.js]: const a

Comment (applmak) [inline client/status/status.js]: single '

File: client/status/status.js [hunk 4]
```diff
@@ -207,15 +219,19 @@ fetchJson('config').then(config => {
...
+      a.id = 'module_' + module.name;
       a.addEventListener('click', function() {
         fetch('/api/play?module=' + module.name, { method: 'POST' });
       });
+      a.textContent = module.name;
+      li.appendChild(a);
+    } else {
+      let m = document.createElement("span");
+      m.textContent = module.name + ' (not in a playlist)';
```
Comment (applmak) [inline client/status/status.js]: `${module.name} (not in a playlist)`;

File: client/status/status.js [hunk 5]
```diff
@@ -207,15 +219,19 @@ fetchJson('config').then(config => {
...
+    if (modulesInPlaylist[module.name] > 1) {
+      let a = document.createElement("a");
+      a.id = 'module_' + module.name;
       a.addEventListener('click', function() {
         fetch('/api/play?module=' + module.name, { method: 'POST' });
       });
+      a.textContent = module.name;
+      li.appendChild(a);
+    } else {
+      let m = document.createElement("span");
```
Comment (applmak) [inline client/status/status.js]: single '

File: client/status/status.js (hunk 1 above)
Comment (plural) [inline client/status/status.js]: done.

File: client/status/status.js (hunk 2 above)
Comment (plural) [inline client/status/status.js]: done.

File: client/status/status.js (hunk 3 above)
Comment (plural) [inline client/status/status.js]: done

Comment (plural) [inline client/status/status.js]: done

File: client/status/status.js (hunk 5 above)
Comment (plural) [inline client/status/status.js]: done

File: client/status/status.js (hunk 4 above)
Comment (plural) [inline client/status/status.js]: done

**Pull request merged**